src/main/java/com/google/devtools/build/lib/bazel/bzlmod/RemoteFile.java

```diff
@@ -0,0 +1,20 @@
+package com.google.devtools.build.lib.bazel.bzlmod;
+import java.net.URL;
+import java.util.List;
+
+/**
+ * For use with IndexRegistry and associated files. A simple pojo to track remote files that are
+ * offered at multiple urls (mirrors) with a single integrity.
+ * We split up the file here to simplify the dependency.
+ */
+class RemoteFile {
```

Comment:
Make this a `record`, then it's short enough to be moved into `IndexRegistry` as an inner class.

Reply:
I can't move it within IndexRegistry because I use it in ArchiveRepoSpecBuilder ; it would cause circular reference.
I can move it into ArchiveRepoSpecBuilder or keep as separate file?

I changed to record -- I wasn't aware Bazel is on a new JDK.
(How can I tell the version?)
 
I also had to write a custom serializer since GSON can't set the record fields as they were final
(I commented on the rationale in the code)

Reply:
Bazel uses Java/JDK 21. You can look for `--java_language_version` in `.bazelrc`. GSON 2.10.1 has support for records.

Reply:
Cool -- I changed one of the tests to use Text blocks also to simplify the JSON processing;
Yay for new Java.